client/state/data-layer/wpcom/account-recovery/lookup/index.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { isString, tap } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import { ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST } from 'state/action-types';
+import {
+	fetchResetOptionsSuccess,
+	fetchResetOptionsError,
+} from 'state/account-recovery/reset/actions';
+
+export const fromApi = data => ( [
+	{
+		email: data.primary_email,
+		sms: data.primary_sms,
+		name: 'primary',
+	},
+	{
+		email: data.secondary_email,
+		sms: data.secondary_sms,
+		name: 'secondary',
+	},
+] );
+
+export const validate = ( { primary_email, primary_sms, secondary_email, secondary_sms } ) => {
+	if ( ! [ primary_email, primary_sms, secondary_email, secondary_sms ].every( isString ) ) {
+		throw Error( 'Unexpected response format from /account-recovery/lookup' );
+	}
+};
+
+export const handleRequestResetOptions = ( { dispatch }, { userData } ) => (
+	wpcom.req.get( {
+		body: userData,
+		apiNamespace: 'wpcom/v2',
+		path: '/account-recovery/lookup',
+	} ).then( data => dispatch( fetchResetOptionsSuccess( fromApi( tap( data, validate ) ) ) ) )
+	.catch( error => dispatch( fetchResetOptionsError( error ) ) )
```

Comment:
if we go ahead and use the HTTP middleware here we won't have to introduce `nock` in the tests

```js
export const storeContactInfo = ( { dispatch }, action, next, data ) =>
	dispatch( fetchResetOptionsSuccess( fromApi( tap( data, validate ) ) ) );

export const reportError = ( { dispatch }, action, next, error ) =>
	dispatch( fetchResetOptionsError( error ) );

export const requestLookup = ( { dispatch }, action ) => dispatch( http( {
	method: 'GET',
	path: '/account-recovery/lookup',
	apiNamespace: 'wpcom/v2',
	body: action.userData,
}, action );

export default {
	[ REQUEST ]: [ dispatchRequest( requestLookup, storeContactInfo, reportError ) ]
}
```